src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -401,3 +392,75 @@ const IndexPatternEditorFlyoutContentComponent = ({
 };
 
 export const IndexPatternEditorFlyoutContent = React.memo(IndexPatternEditorFlyoutContentComponent);
+
+// loadMatchedIndices is called both as an side effect inside of a parent component and the inside forms validation functions
+// that are challenging to synchronize without a larger refactor
+// Use memoizeOne as a caching layer to avoid excessive network requests on each key type
+const loadMatchedIndices = memoizeOne(
```

Comment:
`memoizeOne` is the simplest way I can think of to share the state between validation and component side-effect.

Reply:
I was thinking in refactoring the code code once https://github.com/elastic/kibana/pull/109238 is merged so I am not sure we need to optimise for now.

Reply:
You can see in the documentation I wrote how this will be implemented here

https://github.com/elastic/kibana/pull/109238/files#diff-4ca1882173722332a4f281526d2ac2e11e0ad7fbe3acbf88e1641d58ba859093R296

Reply:
@sebelga, I've just tried to remove it and I get 3 network requests on each keystroke instead of 1 without it. 
Do you think this quick trick doesn't worth it? I also planned to backport to at least 7.15 (looking at it like on a bug). 
We can still refactor in 7.16+ when using your new API. wdyt? 
I am open to reverting, just want to make sure we consider this ^^

Reply:
Cool yes the trick is worth it and we can remove it later on 👍 Surprised it is 3 requests and not 2. But still worth it 😊

Reply:
Looks like 3 are: 

1. the first is coming from validation
2. The side effect inside the parent component 
3. then again from validation for the same string. maybe because the component triggered some re-renders

Reply:
> The side effect inside the parent component

Mmmm. Can you point me to that code? The parent component also triggers a "fetchIndices()"?

Reply:
This is where fetch №2 happens: 
> The side effect inside the parent component
https://github.com/elastic/kibana/blob/0a8d42c80b07e47737ef0cc78162b4e5e34a950d/src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx#L295

Reply:
I see on L286

```js
useEffect(() => {
  reloadMatchedIndices(title);
}, [allowHidden, reloadMatchedIndices, title]);
```

I am not sure I fully understand why we need this. I thought it was the validator inside the titleField that was triggering the reload. Why do we need also this effect?

Reply:
So if I understand correctly, you call twice the `reloadMatchedIndices`? Once inside the validation (when `title` changes) and once in this `useEffect` again when `title` changes? It seems that the `useEffect` is redundant (it also has `allowHidden` dependency which should not be there).

Reply:
> (it also has allowHidden dependency which should not be there).

yes, I agree.

> It seems that the useEffect is redundant

Hm, it is indeed might be redundant. I haven't tried to remove it